### PR TITLE
[FIX] NEW, #6 z-index issue with table panel and source code dialog

### PR DESCRIPTION
**Pull request by thomich**
have found that new code does call editor.selection.getBookmark(); which does bring the panel back.
commenting it our, will help and all works as needed, also position of cursor.

please check my code..

thanks

### CODE REVIEW

**Comment by chadcallahan** [comment]
Doesn't commenting out that line bring [this issue](https://github.com/christiaan/tinymce-codemirror/pull/7) back?

File: plugins/codemirror/source.html [hunk 1]
```diff
@@ -103,7 +103,8 @@
 
 	// Borrowed from codemirror.js themeChanged function. Sets the theme's class names to the html element.
 	// Without this, the background color outside of the codemirror wrapper element remains white.
-	document.documentElement.className += CMsettings.config.theme.replace(/(^|\s)\s*/g, " cm-s-");
```
Comment (chadcallahan) [inline plugins/codemirror/source.html:106]: If this is causing an error, rather than commenting out, why not just do a check on CMsettings.config.theme?

**Comment by thomich** [comment]
your solution for #7 will break solution for #6 .
but #6 doesn't work if solution for #7 is not commented... confused.. 

then my workaround for #2 does only work partially...

think we have to study more deeper ;)...

File: plugins/codemirror/source.html (hunk 1 above)
Comment (thomich) [inline plugins/codemirror/source.html:106]: what you mean with check CMsettings.config.theme?
do we have to init that config in tinyMCE codemirror init part?

Comment (thomich) [inline plugins/codemirror/source.html:106]: did you mean something like this?
`
if(CMsettings.config.theme) {
        document.documentElement.className += CMsettings.config.theme.replace(/(^|\s)\s*/g, " cm-s-");
    }
`

Comment (chadcallahan) [inline plugins/codemirror/source.html:106]: Yes.

File: plugins/codemirror/source.html [hunk 2]
```diff
@@ -169,10 +177,11 @@
 	"use strict";
 
 	var idx = haystack.indexOf(needle), depth = 0, x;
-	for (x = idx; x >= 0; x--) {
+	for (x = idx -1; x >= 0; x--) {
```
Comment (chadcallahan) [inline plugins/codemirror/source.html:182]: Curious; why was the -1 added?

Comment (thomich) [reply plugins/codemirror/source.html:182]: this is about setting cursor in codemirror inside `&nbsp;` or `&uuml;` or any similar entity.
w/o that code and you set cursor inside eg. `&nbsp;` and press ok in codemirror window, `&nbsp;` will be visible in tinymce...

**Comment by thomich** [comment]
whats state of this PR.... anything i/we need to change to optimize?
any suggestions are welcome ;)

regards
thomi

**Comment by christiaan** [comment]
Did you process all feedback and update this PR?

If not, you can fix it on master ;)

Thanks for taking the time!

**Pull request merged**